core/dblookupext/historyRepository.go

```diff
@@ -272,40 +270,42 @@ func (hr *historyRepository) onNotarizedInRegularBlockOfMeta(blockOfMetaNonce ui
 func (hr *historyRepository) onNotarizedMiniblock(metaBlockNonce uint64, metaBlockHash []byte, shardOfContainingBlock uint32, miniblockHeader block.MiniBlockHeader) {
 	miniblockHash := miniblockHeader.Hash
 	isIntra := miniblockHeader.SenderShardID == miniblockHeader.ReceiverShardID
-	notarizedAtSource := miniblockHeader.SenderShardID == shardOfContainingBlock
+	isFromMeta := miniblockHeader.SenderShardID == core.MetachainShardId
 	isToMeta := miniblockHeader.ReceiverShardID == core.MetachainShardId
+	isNotarizedAtBoth := isIntra || isToMeta || isFromMeta
```

Comment:
not sure if isFromMeta is sufficient for the rewards miniblocks. 
maybe notarizedAtDestFromMeta? which should be isNotarizedAtDestination && isFromMeta

Reply:
Fixed as discussed (now notification for rewards is received correctly). Rewards handled as "normal cross-shard" transactions.